app/cloud.go

```diff
@@ -16,34 +18,122 @@ import (
 	"github.com/mattermost/mattermost-server/v6/product"
 	"github.com/mattermost/mattermost-server/v6/shared/i18n"
 	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/mattermost/mattermost-server/v6/store"
 )
 
-func (a *App) NotifySystemAdminsToUpgrade(c *request.Context, currentUserTeamID string) *model.AppError {
-	userId := c.Session().Id
+const lastTrialNotificationTimeStamp = "LAST_TRIAL_NOTIFICATION_TIMESTAMP"
+const lastUpgradeNotificationTimeStamp = "LAST_UPGRADE_NOTIFICATION_TIMESTAMP"
+const defaultCloudNotifyAdminCoolOffDays = 30
 
-	fakeId := strings.ReplaceAll(model.CloudNotifyAdminInfo, "_", "") + "123456"
+// Ensure cloud service wrapper implements `product.CloudService`
+var _ product.CloudService = (*cloudWrapper)(nil)
+
+// cloudWrapper provides an implementation of `product.CloudService` for use by products.
+type cloudWrapper struct {
+	cloud einterfaces.CloudInterface
+}
+
+func (a *App) SaveAdminNotification(userId string, notifyData *model.NotifyAdminToUpgradeRequest) *model.AppError {
+	requiredFeature := notifyData.RequiredFeature
+	requiredPlan := notifyData.RequiredPlan
+	trial := notifyData.TrialNotification
+
+	if a.UserAlreadyNotifiedOnRequiredFeature(userId, requiredFeature) {
```

Comment:
The way it's implemented now, a user can notify once within the cool off period about a feature. Remember, we don't delete any notifications until they have been sent. Sending notifications only happens once within the cool off period.

Reply:
Oops, got confused with the job running every minute for testing and not realizing it only sends the notification once every 30 days no matter what. Looks like there's nothing to do here since we want them to be able to notify after the cool off. Thanks!